src/lib/twitter/notifyTwitter.ts

```diff
@@ -0,0 +1,40 @@
+import TwitterAPI from 'twitter-api-v2';
+import { NFTSale, SaleMethod } from "lib/marketplaces";
+import axios from 'axios';
+import { getFromCache, insertIntoCache } from './mediaIdCache';
+
+export default async function notifyTwitter(twitterClient: TwitterAPI, nftSale: NFTSale) {
+    const nftName = nftSale.nftData?.name;
+    const text = `${nftName} has just been sold ${
+        nftSale.method === SaleMethod.Bid ? "via bidding " : ""
+    }for ${nftSale.getPriceInSOL()} S◎L at ${nftSale.marketplace.name}! #SolanaNFTs #NFTSale`
+    const mediaArr: string[] = [];
+    const imgIdFromCache = getFromCache(nftSale.token);
+    if (imgIdFromCache) {
+        mediaArr.push(imgIdFromCache);
+    } else if (Boolean(nftSale.nftData?.image)) {
+        const data = await getImageDataFromUrl(nftSale.nftData?.image as string);
+        const media = await twitterClient.v1.uploadMedia(data, { type: getDataType(data), shared: true });
+        insertIntoCache(nftSale.token, media);
+        mediaArr.push(media);
+    }
+    return twitterClient.v1.tweet(text, {
+        media_ids: mediaArr
+    })
+}
+
+function getDataType(buffer: Buffer) {
+    const magic = [
```

Comment:
It looks like you can simplify this function by doing this:
```ts
const dataTypesByHexCode = {
  'ffd8ffe0':  'jpg',
  ...
}

const bufferDataType = buffer.toString("hex", 0, 4)

return dataTypesByHexCode[bufferDataType];
```

Where did you find this snippet? It's looks magical. Keen to learn more. 
```
buffer.toString("hex", 0, 4)
```
Have you considered using a more standard way of fetching the image type from buffer? 
https://github.com/sindresorhus/image-type

Reply:
> Where did you find this snippet? It's looks magical. Keen to learn more.
LOL kinda stole it from here: https://stackoverflow.com/a/29964395/12127196

> Have you considered using a more standard way of fetching the image type from buffer?
Yea, I suppose I should, lol. I'll use the `file-type` package instead tho since it supports more file types.